sw/device/lib/testing/clkmgr_testutils.h

```diff
@@ -0,0 +1,41 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_CLKMGR_TESTUTILS_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_CLKMGR_TESTUTILS_H_
+
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+
+/**
+ * Returns the transactional block's clock status.
+ *
+ * @param clkmgr A clkmgr handle.
+ * @param params Hardware instance parameters.
+ * @param clock The transactional clock ID.
+ * @return The transactional block's clock status.
+ */
+bool clkmgr_testutils_get_trans_clock_status(dif_clkmgr_t *clkmgr,
+                                             dif_clkmgr_params_t params,
+                                             dif_clkmgr_hintable_clock_t clock);
+
+/**
+ * Verifies clock gating for transactional clocks.
+ *
+ * For a transactional block, the clock is gated only if the software enables
+ * the clock gating AND the block is idle. This function sets the clock gating
+ * hint bit to 0 and spinwaits until the clock value matches the expected.
+ *
+ * @param clkmgr A clkmgr handle.
```

Comment:
As above, this is a generally useful part of .h files, but if you don't reword it, I'd rather just read the function declaration.